app/controllers/users/confirmations_controller.rb

```diff
@@ -60,33 +66,44 @@ def process_user_with_password_errors
     end
 
     def process_user_with_confirmation_errors
+      return process_already_confirmed_user if @confirmable.confirmed?
+
       set_view_variables
       render :new
     end
 
+    def process_already_confirmed_user
+      action_text = 'Please sign in.' unless user_signed_in?
+      flash[:error] = t('devise.confirmations.already_confirmed', action: action_text)
+
+      redirect_to user_signed_in? ? dashboard_index_url : new_user_session_url
+    end
+
     def process_unconfirmed_user
       set_view_variables
 
       if resource.confirmation_period_expired?
         flash[:error] = UserDecorator.new(resource).confirmation_period_expired_error
```

Comment:
Good question. I think we're fine here, but I don't think it hurts to use `flash.now`. I wonder if we should use `flash.now` everywhere because I don't think we have any use cases where we want the flash message to persist after redirection, right?